Code/GraphMol/MonomerMol/testMonomerMol.cpp

```diff
@@ -0,0 +1,171 @@
+    for (auto atom : mol.atoms()) {
+        chains.insert(getPolymerId(atom));
+    }
+    std::stringstream fasta;
+    bool first = true;
+    for (auto chain_id : chains) {
+        if (!first) {
+            fasta << "\n";
+        }
+        first = false;
+        fasta << ">Chain " << chain_id << "\n"; // add title
+        auto chain = getPolymer(mol, chain_id);
+        for (auto atom : chain.atoms) {
+            // Really there should be a lookup to ensure that these are all 1 letter
+            // And a sort by residue number & connectivity
+            const auto* res_info = static_cast<const RDKit::AtomPDBResidueInfo*>(mol.getAtomWithIdx(atom)->getMonomerInfo());
+            fasta << res_info->getResidueName();
+        }
+    }
+    return fasta.str();
+}
+
+void neutralize_atoms(RDKit::ROMol& mol)
+{
+    // Algorithm for neutralizing molecules from
+    // https://www.rdkit.org/docs/Cookbook.html#neutralizing-molecules by Noel
+    // O’Boyle Will neutralize the molecule by adding or removing hydrogens as
+    // needed. This will ensure SMILES can be used to match atomistic structures
+    // to the correct monomer.
+    static const std::unique_ptr<RDKit::RWMol> neutralize_query(
+        RDKit::SmartsToMol(
+            "[+1!h0!$([*]~[-1,-2,-3,-4]),-1!$([*]~[+1,+2,+3,+4])]"));
+    for (const auto& match : RDKit::SubstructMatch(mol, *neutralize_query)) {
+        auto atom = mol.getAtomWithIdx(match[0].second);
+        auto chg = atom->getFormalCharge();
+        auto hcount = atom->getTotalNumHs();
+        atom->setFormalCharge(0);
+        atom->setNumExplicitHs(hcount - chg);
+        atom->updatePropertyCache();
+    }
+}
+
+TEST_CASE("FASTAConversions") {
+  SECTION("SIMPLE") {
+    // Build MonomerMol with a single chain
+    RWMol monomer_mol;
+    addMonomer(monomer_mol, "R", 1, "PEPTIDE1");
+    addMonomer(monomer_mol, "D");
+    addMonomer(monomer_mol, "K");
+    addMonomer(monomer_mol, "I");
+    addMonomer(monomer_mol, "T");
+
+    addConnection(monomer_mol, 0, 1, ConnectionType::FORWARD);
+    addConnection(monomer_mol, 1, 2, ConnectionType::FORWARD);
+    addConnection(monomer_mol, 2, 3, ConnectionType::FORWARD);
+    addConnection(monomer_mol, 3, 4, ConnectionType::FORWARD);
+
+    CHECK(std::string(">Chain PEPTIDE1\nRDKIT") == to_fasta(monomer_mol));
+  }
+
+  SECTION("MultipleChains") {
+    // Build MonomerMol with two chains
+    RWMol monomer_mol;
+    auto midx1 = addMonomer(monomer_mol, "R", 1, "A");
+    auto midx2 = addMonomer(monomer_mol, "D");
+
+    auto midx3 = addMonomer(monomer_mol, "K", 1, "B");
+    auto midx4 = addMonomer(monomer_mol, "I");
+    auto midx5 = addMonomer(monomer_mol, "T");
+
+    addConnection(monomer_mol, midx1, midx2, ConnectionType::FORWARD);
+    addConnection(monomer_mol, midx3, midx4, ConnectionType::FORWARD);
+    addConnection(monomer_mol, midx4, midx5, ConnectionType::FORWARD);
+
+    CHECK(std::string(">Chain A\nRD\n>Chain B\nKIT") == to_fasta(monomer_mol));
```

Comment:
This is a good integration-type test, but I think it would also be useful to add a check of the number of atoms and bonds in `monomer_mol`